### DISCUSSION TITLE
How to Store multiple values in a column?

### DISCUSSION THREAD

**Discussion by maheshj01** [supabase/supabase#1878] [Feature Requests]
I would like to add a column to my table that takes in a set of values like an array. But in the available set of types, I don't see any type that allows adding multiple values in a row.
Available types that I See.

```
- bool
- date
- float4
- float8
- int2
- int8
- json
- jsonb
- text
- time
- timestamp
- timestampz
- timetz
- uuid
- varchar
```
E.g I have a table which has columns  words,meanings, synonyms
The column synonym can have multiple values 

Heres what I am expecting

|Words   | Meanings  |Synonyms   | something   |something   |
|---|---|---|---|---|
|word1   | Meaning1  | [synonym1,synonym2]   |   |   |
|word2  | Meaning2  |[synonym1] |  |   |
|word3   | Meaning3 | [synonym1,synonym2,Synonym3] | |   |

I do understand there are hackable ways like storing comma-separated-values, But I am curious if multiple values are allowed if not if there are any plans to add them in the future? or is there any limitation with the database. Sorry if my question is irrelevant I really don't have much idea about the databases.

**Comment by soedirgo** [comment]
Hey @maheshmnj, does an array of `text` fulfill your needs?

<img width="635" alt="Screenshot 2021-06-08 at 9 49 02 PM" src="https://user-images.githubusercontent.com/31685197/121196941-79958a80-c8a3-11eb-83f9-e1cd2acee63d.png">

**Reply by maheshj01** [reply]
Hey thanks not sure how I missed that

**Reply by maheshj01** [reply]
This might seem a bit off-topic I am actually importing the data from an excel sheet and one of the columns (synonym in the above example) is stored as a comma-separated text in excel, I am importing that sheet into superbase. But what I would like to do is import that particular column in the form of an array. Currently, it imports it as a normal text (Which is correct but not what I expect).

**Reply by soedirgo** [reply]
Can you try running:
```sql
alter table your_table alter synonyms set data type text[] using regexp_split_to_array(synonyms, ',')
```
after importing the CSV?

**Reply by maheshj01** [reply]
where do I run this query?

**Reply by maheshj01** [reply]
> Can you try running:
> 
> ```sql
> alter table your_table alter synonyms set data type text[] using regexp_split_to_array(synonyms, ',')
> ```
> 
> after importing the CSV?

anything is fine either modifying the excel sheet before importing or after importing modifying the table.
But modifying excel is daunting because it has around 500+ rows and may I know what format should be the cell in order to be detected as array(TEXT) in the database?

**Reply by soedirgo** [reply]
You can run this in the SQL editor (SQL in the sidebar) after importing it through the Table editor (Table editor > New table > Add existing content > Upload CSV).

> may I know what format should be the cell in order to be detected as array(TEXT) in the database?

I think the format is all `text` when importing from CSV, but I'm not sure. It should be fine if you just leave it as is.

**Reply by maheshj01** [reply]
> You can run this in the SQL editor (SQL in the sidebar) after importing it through the Table editor (Table editor > New table > Add existing content > Upload CSV).
> 
Boom 🚀 that worked like a charm and saved my lot of time thank you so much

**Comment by himeshp** [comment]
is this possible with UI? the array of text would be basically a list of entries from another table.
it will be great, once we apply the relationship user can just select from the list of dropdowns, currently either user types it which is prone to error. any thoughts here?

**Reply by himeshp** [reply]
can anyone from the supabase team guide me here?
